### PR TITLE
Import api modules from local-dev-lib: 2

### PR DESCRIPTION
**Pull request by brandenrodgers**
## Description and Context

Updating api module imports to pull from local-dev-lib.

Updating imports for:
- local-dev-lib/api/projects (formerly `cli-lib/api/dfs`)
- local-dev-lib/api/functions
- local-dev-lib/api/secrets
- local-dev-lib/api/marketplaceValidation

## TODO

- Test this thoroughly to make sure everything works. Mainly error handling.

### CODE REVIEW

File: packages/cli/lib/projectsWatch.js
Comment (camden11) [inline packages/cli/lib/projectsWatch.js:15]: Do we need to figure this out before merging?

Comment (brandenrodgers) [reply packages/cli/lib/projectsWatch.js:15]: Thanks for catching this. I think I need to get this stuff ported to LDL before we can merge. I had forgotten about this.

Comment (brandenrodgers) [reply packages/cli/lib/projectsWatch.js:15]: Ported [here](https://github.com/HubSpot/hubspot-local-dev-lib/pull/104), but still needs testing. I'll do that tomorrow

**Review by camden11 — APPROVED** [review]
Good to go once the local dev lib changes are out!

**Pull request merged**
